fabric-data-generation-api-v1/src/testmod/java/net/fabricmc/fabric/test/datagen/DataGeneratorTestEntrypoint.java

```diff
@@ -163,11 +182,51 @@ public void generateBlockStateModels(BlockStateModelGenerator blockStateModelGen
 			blockStateModelGenerator.registerSimpleCubeAll(SIMPLE_BLOCK);
 			blockStateModelGenerator.registerSimpleCubeAll(BLOCK_WITHOUT_ITEM);
 			blockStateModelGenerator.registerSimpleCubeAll(BLOCK_WITHOUT_LOOT_TABLE);
+
+			// TODO: needs more simplification?
+			BlockModelBuilder customModel = BlockModelBuilder.createNew(new Identifier(MOD_ID, "custom"))
+					.addTexture("texture1", new Identifier(MOD_ID, "block_with_custom_model_1_1"))
+					.addTexture("texture2", new Identifier(MOD_ID, "block_with_custom_model_1_2"))
+					.addDisplay(DisplayBuilder.Position.FIXED, new DisplayBuilder()
+							.rotate(45, 45, 45)
+							.scale(2))
+					.addElement(new ElementBuilder(new Vector3d(1.2), new Vector3d(14.8))
+							.addFace(Direction.NORTH, new FaceBuilder("texture1")
+									.withUv(4, 6, 15, 3)
+									.withRotation(VariantSettings.Rotation.R90)
+									.withCulling(Direction.NORTH)
+									.withTintIndex(3))
+							.withRotation(new RotationBuilder(1, 2, 3, Direction.Axis.X, RotationBuilder.Angle.PLUS22_5)
+									.rescale(true))
+							.withShading(false))
+					.occludes(false);
+			blockStateModelGenerator.buildWithSingletonState(BLOCK_WITH_CUSTOM_MODEL_1, customModel);
+
+			customModel.clearDisplays().clearElements()
+					.addTexture("texture1", new Identifier(MOD_ID, "block_with_custom_model_2_1"))
+					.addTexture("texture2", new Identifier(MOD_ID, "block_with_custom_model_2_2"))
+					.addTexture("texture3", new Identifier(MOD_ID, "block_with_custom_model_2_3"))
+					.occludes(true);
+			blockStateModelGenerator.buildWithSingletonState(BLOCK_WITH_CUSTOM_MODEL_2, customModel);
+
+			blockStateModelGenerator.registerEmptyModel(BLOCK_WITH_EMPTY_MODEL);
 		}
 
 		@Override
 		public void generateItemModels(ItemModelGenerator itemModelGenerator) {
-			//itemModelGenerator.register(item, Models.SLAB);
+			ItemModelBuilder customModel = ItemModelBuilder.copyFrom(Models.GENERATED, TextureMap.layer0(ITEM_WITH_CUSTOM_MODEL_1))
+					.addOverride(new OverrideBuilder(new Identifier(MOD_ID, "overridden_item_model"))
+							.predicate(new Identifier(MOD_ID, "predicate_1"), 0.3f)
+							.predicate(new Identifier(MOD_ID, "predicate_2"), 0.65f))
+					.setGuiLight(ItemModelBuilder.GuiLight.SIDE);
+			itemModelGenerator.build(ITEM_WITH_CUSTOM_MODEL_1, customModel);
+
+			customModel.clearOverrides().setGuiLight(null)
```

Comment:
Again here, reusing the builder is a little odd imo? Doesnt quite follow the builder design pattern too well. Nither does the `clearOverrides()` like methods either 🤔